ksqldb-engine/src/main/java/io/confluent/ksql/logicalplanner/LogicalPlan.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.logicalplanner;
+
+import com.google.common.collect.ImmutableSet;
+import io.confluent.ksql.logicalplanner.nodes.Node;
+import io.confluent.ksql.name.SourceName;
+import java.util.Objects;
+
+public class LogicalPlan {
```

Comment:
Just so I know the strategy, this class will be extended as the new query planner adds handling for limits, joins, etc, right?

Reply:
Right. It's the "container" class of the new logical plan. The plan itself with be a graph of `Nodes` so we will add `JoinNode`, `LimitNode` etc